worker/sshserver/server_test.go

```diff
@@ -0,0 +1,101 @@
+// Copyright 2025 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package sshserver_test
+
+import (
+	"crypto/rand"
+	"crypto/rsa"
+
+	"github.com/juju/testing"
+	"go.uber.org/mock/gomock"
+	"golang.org/x/crypto/ssh"
+	"google.golang.org/grpc/test/bufconn"
+	gc "gopkg.in/check.v1"
+
+	"github.com/juju/juju/worker/sshserver"
+	"github.com/juju/juju/worker/sshserver/mocks"
+)
+
+type sshServerSuite struct {
+	testing.IsolationSuite
+
+	userSigner ssh.Signer
+}
+
+var _ = gc.Suite(&sshServerSuite{})
+
+func (s *sshServerSuite) SetUpSuite(c *gc.C) {
+	s.IsolationSuite.SetUpSuite(c)
+
+	// Setup user signer
+	userKey, err := rsa.GenerateKey(rand.Reader, 2048)
+	c.Assert(err, gc.IsNil)
+
+	userSigner, err := ssh.NewSignerFromKey(userKey)
+	c.Assert(err, gc.IsNil)
+
+	s.userSigner = userSigner
+}
+
+func (s *sshServerSuite) TestSSHServer(c *gc.C) {
+	ctrl := gomock.NewController(c)
+	defer ctrl.Finish()
+
+	mockLogger := mocks.NewMockLogger(ctrl)
+
+	// Firstly, start the server on an in-memory listener
+	listener := bufconn.Listen(8 * 1024)
+	server, err := sshserver.NewServerWorker(sshserver.ServerWorkerConfig{
+		Logger: mockLogger,
+	}, false)
+	c.Assert(err, gc.IsNil)
+
```

Comment:
On this line add the following:

https://github.com/juju/juju/blob/104d6edcbb9916994740d385822a9bd60c05bf61/worker/changestream/worker_test.go#L75